### PR TITLE
Chapter4: CRUD

### PR DESCRIPTION
**Pull request by benjtinsley**
using this book: https://www.manning.com/books/rails-4-in-action

this feature allows users to list created projects for a ticketing app. also allows for projects to be updated and deleted, as well as sets up a redirect if the project does not exist.

other lessons:
- using factory girl to set up factories for testing purposes
- precompiling assets and configuring settings to view the site in production mode to see errors without the stack trace

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -54,6 +54,7 @@ end
 group :test do
   # Adds support for Capybara system testing and selenium driver
   gem 'capybara', '>= 2.15'
+  gem "factory_girl_rails", "~> 4.5"
```
Comment (efatsi) [inline Gemfile]: please read this incredibly long comment chain and then upgrade this gem to `factory_bot` so you don't offend anyone with your codebase

https://github.com/thoughtbot/factory_bot/issues/921

File: app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -18,12 +21,34 @@ def create
...
+    redirect_to projects_path
   end
   
   private
   
+  def set_project
+    @project = Project.find(params[:id])
+  rescue ActiveRecord::RecordNotFound
+    flash[:alert] = "The project you were looking for could not be found."
+    redirect_to projects_path
```
Comment (efatsi) [inline app/controllers/projects_controller.rb]: a more common approach to this is to set a high-level `rescue_from` in your application controller. Benefit is that you can then just use `find(params[:id])` all over the place and know that unfound things will be 404d: Here's shure as an example: https://github.com/vigetlabs/shure/blob/master/app/controllers/application_controller.rb#L10 + https://github.com/vigetlabs/shure/blob/master/app/controllers/application_controller.rb#L73-L77

File: Gemfile (hunk 1 above)
Comment (benjtinsley) [reply Gemfile]: ha i saw the deprecation warnings in the terminal and knew someone would say something. good to know the underlying reason for the name change

File: app/controllers/projects_controller.rb (hunk 2 above)
Comment (benjtinsley) [reply app/controllers/projects_controller.rb]: that makes sense, in this case of having the rescue on the class controller, does it ensure you give the user more specific feedback to their URL. for example in this case, it tells them "the project... could not be found." can you give similarly specific feedback from the application level?

Comment (efatsi) [reply app/controllers/projects_controller.rb]: correct, this is the only way to have type-specific error messages.

**Pull request closed without merging**